apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -337,11 +339,17 @@ impl Plugin for Telemetry {
                             span.record("graphql.operation.name", operation_name);
                         }
                         match (&operation_kind, &operation_name) {
-                            (Ok(Some(kind)), Ok(Some(name))) => {
-                                span.record("otel.name", format!("{kind} {name}"))
+                            (Ok(Some(kind)), Ok(Some(name))) => span.set_span_dyn_attribute(
+                                "otel.name".into(),
```

Comment:
NIT: could be a constant

Reply:
[fbda2bf](https://github.com/apollographql/router/pull/5386/commits/fbda2bfcbb3120f8e0fd74af360fbad140174b7a)